Prefabs/Interactions/Controllables/SharedResources/Scripts/Drive.cs

```diff
@@ -0,0 +1,269 @@
+        /// </summary>
+        public float Value => CalculateValue(baseFacade.DriveAxis, DriveLimits);
+        /// <summary>
+        /// The current normalized value for the drive control between the set limits.
+        /// </summary>
+        public float NormalizedValue => Mathf.InverseLerp(DriveLimits.minimum, DriveLimits.maximum, Value);
+        /// <summary>
+        /// The current step value for the drive control.
+        /// </summary>
+        public float StepValue => CalculateStepValue(baseFacade);
+        /// <summary>
+        /// The current normalized step value for the drive control between the set step range.
+        /// </summary>
+        public float NormalizedStepValue => Mathf.InverseLerp(baseFacade.stepRange.minimum, baseFacade.stepRange.maximum, StepValue);
+
+        /// <summary>
+        /// The calculated direction for the drive axis.
+        /// </summary>
+        public Vector3 AxisDirection
+        {
+            get;
+            protected set;
+        }
+
+        /// <summary>
+        /// The calculated limits for the drive.
+        /// </summary>
+        public FloatRange DriveLimits
+        {
+            get;
+            protected set;
+        }
+
+        /// <summary>
+        /// The base <see cref="DriveFacade"/> for the fundamental drive settings.
+        /// </summary>
+        protected DriveFacade baseFacade;
+        /// <summary>
+        /// The previous state of <see cref="Value"/>.
+        /// </summary>
+        protected float previousValue = float.MaxValue;
+        /// <summary>
+        /// The previous state of <see cref="StepValue"/>.
+        /// </summary>
+        protected float previousStepValue = float.MaxValue;
+        /// <summary>
+        /// The previous state of whether the target value has been reached.
+        /// </summary>
+        protected bool previousTargetValueReached = false;
+
+        /// <summary>
+        /// Sets the target value of the drive to the given normalized value.
+        /// </summary>
+        /// <param name="newValue">The normalized value to set the Target Value to.</param>
+        public abstract void SetTargetValue(float newValue);
+        /// <summary>
+        /// Processes the speed in which the drive can affect the control.
+        /// </summary>
+        /// <param name="driveSpeed">The speed to drive the control at.</param>
+        /// <param name="moveToTargetValue"><see langword="true"/> will allow the drive to automatically move the control to the desired target value.</param>
+        public abstract void ProcessDriveSpeed(float driveSpeed, bool moveToTargetValue);
+
+        /// <summary>
+        /// Gets the axis direction for the given <see cref="DriveAxis"/>.
+        /// </summary>
+        /// <param name="driveAxis">The desired world axis.</param>
+        /// <returns>The direction of the drive axis.</returns>
+        public static Vector3 GetAxisDirection(DriveAxis driveAxis)
```

Comment:
It could, but `DriveAxis` is just an enum on the Drive class. Feels a bit overkill having a whole new class with a single extension method for an enum that's in the same class as this existing method.

Reply:
👍